webnn.json

```diff
@@ -474,55 +474,26 @@
       },
       {
         "name": "build",
-        "args": [
-          {"name": "named operands", "type": "named operands"},
-          {"name": "callback", "type": "build graph callback"},
-          {"name": "userdata", "type": "void", "annotation": "*"}
-        ]
-      },
-      {
-        "name": "buildSync",
         "returns": "graph",
         "args": [
           {"name": "named operands", "type": "named operands"}
         ]
       }
     ]
   },
-  "build graph callback": {
-    "category": "callback",
-    "args": [
-      {"name": "status", "type": "build graph status"},
-      {"name": "graph", "type": "graph"},
-      {"name": "message", "type": "char", "annotation": "const*", "length": "strlen"},
-      {"name": "userdata", "type": "void", "annotation": "*"}
-    ]
-  },
-  "build graph status": {
-    "category": "enum",
-    "values": [
-        {"value": 0, "name": "success"},
-        {"value": 1, "name": "error"},
-        {"value": 2, "name": "context lost"},
-        {"value": 3, "name": "unknown"}
-    ]
-  },
   "input": {
     "category": "structure",
     "members": [
-      {"name": "buffer", "type": "void", "annotation": "const*", "length": "size"},
-      {"name": "size", "type": "size_t"},
-      {"name": "dimensions", "type": "int32_t", "annotation": "const*", "length": "dimensions count", "optional": true},
+      {"name": "resource", "type": "resource"},
+      {"name": "dimensions", "type": "int32_t", "annotation": "const*", "length": "dimensions count"},
       {"name": "dimensions count", "type": "uint32_t", "default": 0}
     ]
   },
-  "output": {
+  "resource": {
     "category": "structure",
     "members": [
-      {"name": "buffer", "type": "void", "annotation": "*", "length": "size", "optional": true},
-      {"name": "size", "type": "size_t"},
-      {"name": "dimensions", "type": "int32_t", "annotation": "const*", "length": "dimensions count", "optional": true},
-      {"name": "dimensions count", "type": "uint32_t", "default": 0}
+      {"name": "buffer", "type": "void", "annotation": "*", "length": "size"},
+      {"name": "size", "type": "size_t"}
```

Comment:
And we may need to support `byteOffset`, such as
```
{"name": "byte offset", "type": "size_t", "default", 0}
```

With it, we read and write data to the buffer starting from `byteOffset`. It would make the node.js code simpler.

Thoughts?